src/test/java/lotto_tests/LottoPurchaseTicketTests.java

```diff
@@ -0,0 +1,103 @@
+package lotto_tests;
+
+import lotto.model.LottoPurchaseTicket;
+import lotto.model.LottoResult;
+import lotto.model.WinningLottoTicket;
+import lotto.model.wrapper.LottoNumber;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.*;
+
+@DisplayName("로또 한장 테스트")
+public class LottoPurchaseTicketTests {
+
+    @DisplayName("로또 생성 테스트")
+    @Test
+    public void generateLottoTest() {
+        assertThatCode(LottoPurchaseTicket::newInstance).doesNotThrowAnyException();
+    }
+
+    @DisplayName("로또 생성 오류 테스트")
+    @ParameterizedTest
+    @NullAndEmptySource
+    @MethodSource("generateLottoExceptionTestCases")
+    public void generateLottoExceptionTest(Set<LottoNumber> numbers) {
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> LottoPurchaseTicket.newInstance(numbers))
+                .withMessageContaining("Lotto Ticket must have six distinct number.");
+    }
+
+    @DisplayName("로또 결과 확인 테스트")
+    @ParameterizedTest
+    @MethodSource("checkLottoTicketTestCases")
+    public void checkLottoTicketTest(LottoPurchaseTicket lottoPurchaseTicket, WinningLottoTicket winningLottoTicket, LottoResult lottoResult) {
+        assertThat(lottoPurchaseTicket.check(winningLottoTicket)).isEqualTo(lottoResult);
+    }
+
+    private static Stream<Arguments> generateLottoExceptionTestCases() {
+        return Stream.of(
+                Arguments.of(new HashSet<>(Arrays.asList(
+                        LottoNumber.of(1), LottoNumber.of(2), LottoNumber.of(3),
+                        LottoNumber.of(4), LottoNumber.of(5), LottoNumber.of(6), LottoNumber.of(7)))),
+                Arguments.of(new HashSet<>(Arrays.asList(
+                        LottoNumber.of(1), LottoNumber.of(2),
+                        LottoNumber.of(3), LottoNumber.of(4))))
+        );
+    }
+
+    private static Stream<Arguments> checkLottoTicketTestCases() {
+        LottoPurchaseTicket lottoPurchaseTicket = LottoPurchaseTicket.newInstance(convertToLottoNumbers(Arrays.asList(1, 2, 3, 4, 5, 6)));
```

Comment:
생성자의 수가 늘어날수록 클래스를 사용하기 위한 유연성은 높아집니다.

```suggestion
        LottoPurchaseTicket lottoPurchaseTicket = LottoPurchaseTicket.newInstance(1, 2, 3, 4, 5, 6);
```